mat/format.go

```diff
@@ -427,58 +382,63 @@ func formatPython(m Matrix, prefix string, _ int, _ byte, squeeze bool, fs fmt.S
 		}
 
 		for j := 0; j < cols; j++ {
-			v := m.At(i, j)
-			if c == 'v' {
-				buf = strconv.AppendFloat(buf[:0], v, 'g', prec, 64)
-			} else {
-				buf = strconv.AppendFloat(buf[:0], v, byte(c), prec, 64)
-			}
-			if fs.Flag('-') {
-				fs.Write(buf)
-				fs.Write(pad[:widths.width(j)-len(buf)])
-			} else {
-				fs.Write(pad[:widths.width(j)-len(buf)])
-				fs.Write(buf)
-			}
+			f = m.RevalueFormatter(f, i, j)
+			f.Format(ws.At(j), c)
 
 			if j < cols-1 {
-				fs.Write([]byte{','})
-				fs.Write(pad[:1])
+				fmt.Fprint(fs, ", ")
 			}
 		}
 
 		fmt.Fprint(fs, el)
 	}
 }
 
-// This is horrible, but it's what we have.
-func fmtString(fs fmt.State, c rune, prec, width int) string {
-	var b strings.Builder
-	b.WriteByte('%')
-	for _, f := range "0+- " {
-		if fs.Flag(int(f)) {
-			b.WriteByte(byte(f))
-		}
-	}
-	if width >= 0 {
-		fmt.Fprint(&b, width)
-	}
-	if prec >= 0 {
-		b.WriteByte('.')
-		if prec > 0 {
-			fmt.Fprint(&b, prec)
-		}
+// state satisfies fmt.State and may varies its responses to Width.
+type state struct {
+	write     func(b []byte) (n int, err error)
+	width     func() (wid int, ok bool)
+	precision func() (prec int, ok bool)
+	flag      func(c int) bool
+
+	w widther
+	p int
+}
+
+var _ fmt.State = (*state)(nil)
+
+// newState return a new state, shadowing given fmt.State fs.
+func newState(fs fmt.State) *state {
```

Comment:
It's not clear to me why this needs to return a pointer type; the fit method returns the mutated `state`.  Either it can mutate in place and not bother returning the value (and having the remaining methods on the pointer), or preferably, returning the mutated non-pointer `state` (this would require that the `fmt.State` assignability check above changes).